### PR TITLE
Update isar dependency to v1.0+

### PR DESCRIPTION
**Pull request by apgapg**
Signed-off-by: Ayush P Gupta <ayushpguptaapg@gmail.com>

## Description

This PR upgrades `isar` dependency to latest so as to make it run on `linux` host.

Please note this is a breaking change in terms of isar and its db path.

Fixes #(issue)

#5 

## Screenshots (if appropriate):

Screenshot from 2022-01-08 07-42-13

## Type of change

Please tick the relevant option by putting an X inside the bracket

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] All current GitHub actions pass
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document

### CODE REVIEW

**Comment by apgapg** [comment]
Any updates?

File: lib/backend/database/helper.dart [hunk 1]
```diff
@@ -6,7 +6,7 @@ import 'package:files/backend/providers.dart';
 class EntityStatCacheHelper {
   Future<EntityStat> get(String path) async {
     final EntityStat? stat =
-        isar.entityStats.where().pathEqualTo(path).findFirstSync();
+        isar.entityStats.where().pathNotEqualTo(path).findFirstSync();
```
Comment (dvd-bnc) [inline lib/backend/database/helper.dart]: why this logic change? i checked the generated file and the pathEqualTo function exists, did i get this wrong originally?

File: lib/backend/providers.dart [hunk 2]
```diff
@@ -42,7 +45,15 @@ class _ProvidersSingleton {
 
   Future<void> _init() async {
     if (_inited) return;
-    _isar = await openIsar();
+    WidgetsFlutterBinding.ensureInitialized();
+    final dir = await getApplicationDocumentsDirectory();
```
Comment (dvd-bnc) [inline lib/backend/providers.dart:49]: was the old database inited in the same dir? with the same name? i would like to keep retro compatibility

File: lib/backend/providers.dart [hunk 3]
```diff
@@ -42,7 +45,15 @@ class _ProvidersSingleton {
...
+    _isar = await Isar.open(
+      schemas: [EntityStatSchema],
+      directory: p.join(
+        dir.path,
+        'isar',
```
Comment (dvd-bnc) [inline lib/backend/providers.dart]: nit: removing this trailiing comma imo would look better

**Review by dvd-bnc — COMMENTED** [review]
Thanks for the pr! Sorry if i got to look at it only now but i didn't get notified in any way, before merging i'd like at least an explaination for the comments, but for the rest it looks good!

File: lib/backend/database/helper.dart (hunk 1 above)
Comment (apgapg) [reply lib/backend/database/helper.dart]: I apologize, somehow i made this a breaking change. correcting this.
The reason might be I am not aware of `isar` package fully.

File: lib/backend/providers.dart (hunk 2 above)
Comment (apgapg) [reply lib/backend/providers.dart:49]: I see openIsar now doesnt exist in v1.0, hence we need to change its initialisation way.
In docs too author mentioned 
```
Breaking 
The internal database format has been changed to improve performance. Old databases do not work anymore!
```
hence backward compatibility might not be possible.

Going forward, we need to check if we are defining path of isar correctly as per we have decided or not.

File: lib/backend/providers.dart (hunk 3 above)
Comment (apgapg) [reply lib/backend/providers.dart]: done. I see that was unnecessary as line length was < 80

**Comment by apgapg** [comment]
@HrX03 except for db initialisation, I have resolved other comments. Requesting for review again

**Review by dvd-bnc — APPROVED** [review]
Looks good now, should be good to go for me

**Pull request merged**